Makefile.am

```diff
@@ -50,3 +50,10 @@ dist_noinst_DATA = \
 	AUTHORS COPYING README \
 	example.c \
 	testinput.c
+
+if CAN_COMPILE_EXAMPLE
+EXTRA_PROGRAMS = example
+example_SOURCES = example.c
+example_LDADD = $(LDADD) $(LIBGCRYPT_LIBS) $(lib_LTLIBRARIES)
```

Comment:
I meant to simply append the LDADD line you already had (with the .la files) with the LIBGCRYPT_LIBS